File: src/dotnet/APIView/APIViewWeb/Models/ProjectModel.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using APIViewWeb.LeanModels;

namespace APIViewWeb.Models;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ProjectChangeAction
{
    Created,
    Edited,
    Deleted,
    UnDeleted,
    ReviewLinked,
    ReviewUnlinked,
    NamespaceStatusChanged,
}

public class Project
{
    [JsonPropertyName("id")]
    public string Id { get; set; }
    public string CrossLanguagePackageId { get; set; }
    public string DisplayName { get; set; }
    public string Description { get; set; }
    public List<string> Owners { get; set; }
    public string Namespace { get; set; }
    public Dictionary<string, PackageInfo> ExpectedPackages { get; set; }
    public ProjectNamespaceInfo NamespaceInfo { get; set; }
    public List<ProjectChangeHistory> ChangeHistory { get; set; }
    // Maps language name (e.g. "Python", "JavaScript") to the associated review ID.
    public Dictionary<string, string> Reviews { get; set; } = new (StringComparer.OrdinalIgnoreCase);
    public HashSet<string> HistoricalReviewIds { get; set; }
    public DateTime CreatedOn { get; set; }
    public DateTime LastUpdatedOn { get; set; }
    public bool IsDeleted { get; set; }
}

public class ProjectChangeHistory : ChangeHistoryModel
{
    public ProjectChangeAction ChangeAction { get; set; }
}

public class PackageInfo
{
    public string PackageName { get; set; }
    public string Namespace { get; set; }
}

File: src/dotnet/APIView/ClientSPA/src/app/_models/projectModel.ts
import { ProjectNamespaceInfo } from './namespaceModel';

export interface ProjectChangeHistory {
  changedBy: string;
  changedOn: string;
  notes: string;
  changeAction: ProjectChangeAction;
}

export enum ProjectChangeAction {
  Created = 'Created',
  Edited = 'Edited',
  Deleted = 'Deleted',
  UnDeleted = 'UnDeleted',
  ReviewLinked = 'ReviewLinked',
  ReviewUnlinked = 'ReviewUnlinked',
  NamespaceStatusChanged = 'NamespaceStatusChanged'
}

export interface PackageInfo {
  packageName: string;
  namespace: string;
}

export interface Project {
  id: string;
  crossLanguagePackageId: string;
  displayName: string;
  description: string;
  owners: string[];
  namespace: string;
  expectedPackages: { [language: string]: PackageInfo };
  namespaceInfo: ProjectNamespaceInfo;
  changeHistory: ProjectChangeHistory[];
  reviews: { [language: string]: string };
  historicalReviewIds: string[];
  createdOn: string;
  lastUpdatedOn: string;
  isDeleted: boolean;
}

export interface RelatedReviewItem {
  id: string;
  packageName: string;
  language: string;
  subscribers: string[];
  isClosed: boolean;
  isApproved: boolean;
  packageType: string | null;
  namespaceReviewStatus: string;
  createdBy: string;
  createdOn: string;
  lastUpdatedOn: string;
  isDeleted: boolean;
  projectId: string;
  crossLanguagePackageId: string;
}

export interface RelatedReviewsResponse {
  currentReviewId: string;
  projectId: string;
  projectName: string;
  crossLanguagePackageId: string;
  reviews: RelatedReviewItem[];
}
